react_juce/core/TextInputView.cpp

```diff
@@ -0,0 +1,225 @@
+
+    void TextInput::textEditorReturnKeyPressed(juce::TextEditor &)
+    {
+        invokeChangeEventIfNeeded();
+    }
+
+    void TextInput::textEditorFocusLost(juce::TextEditor &)
+    {
+        invokeChangeEventIfNeeded();
+    }
+
+    void TextInput::invokeChangeEventIfNeeded()
+    {
+        if (dirty)
+        {
+            // Invoke JavaScript's `change` event.
+            if ((*props).contains(TextInputView::onChangeProp) && (*props)[TextInputView::onChangeProp].isMethod())
+            {
+                std::array<juce::var, 1> args{{detail::makeChangeEventObject(getText())}};
+                juce::var::NativeFunctionArgs nfArgs(juce::var(), args.data(), static_cast<int>(args.size()));
+                std::invoke((*props)[TextInputView::onChangeProp].getNativeFunction(), nfArgs);
+            }
+            dirty = false;
+        }
+    }
+
+    //==============================================================================
+
+    TextInputView::TextInputView()
+        : textInput(&props)
+    {
+        addAndMakeVisible(textInput);
+        textInput.addListener(&textInput);
+        textInput.setPopupMenuEnabled(false);
+    }
+
+    void TextInputView::setProperty(const juce::Identifier &name, const juce::var &value)
+    {
+        View::setProperty(name, value);
+        if (name == valueProp)
+        {
+            if (!value.isString())
+                throw std::invalid_argument("Invalid prop value. Prop \'value\' must be a string.");
+            textInput.setControlled(true);
+            textInput.setControlledValue(value);
+        }
+        if (name == placeholderProp)
+        {
+            if (!value.isString())
+                throw std::invalid_argument("Invalid prop value. Prop \'placeholder\' must be a string.");
+            textInput.setPlaceholderText(value);
+        }
+        if (name == placeholderColorProp)
+        {
+            if (!value.isString())
+                throw std::invalid_argument("Invalid prop value. Prop \'placeholder-color\' must be a color string.");
+            juce::String hexPlaceholderColor = value;
+            juce::Colour placeholderColor = juce::Colour::fromString(hexPlaceholderColor);
+            textInput.setPlaceholderColour(placeholderColor);
+        }
+        if (name == maxlengthProp)
+        {
+            if (!value.isDouble())
+              throw std::invalid_argument("Invalid prop value. Prop \'maxlength\' must be a number.");
+            textInput.setMaxLength(value);
+        }
+        if (name == readonly)
+        {
+            textInput.setReadOnly(value);
+        }
+
+        textInput.applyFontToAllText(getFont());
+
+        juce::String hexColor = props.getWithDefault(colorProp, "ff000000");
+        juce::Colour colour = juce::Colour::fromString(hexColor);
+        textInput.applyColourToAllText(colour);
+
+        int just = props.getWithDefault(justificationProp, 1);
```

Comment:
Super nitpicky but could make this `const int just`

Reply:
I fixed it in c724958.